### PR TITLE
config: move all travis commands to .ci folder to ease execution on local

### PR DESCRIPTION
**Pull request by romani**
this will help to investigate weird PR validation error that is identified at https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/632#issuecomment-355160331.

### CODE REVIEW

**Comment by romani** [comment]
Travis failure:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (default) on project sevntu-checkstyle-sonar-plugin: Execution default of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check failed: String index out of range: -1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check (default) on project sevntu-checkstyle-sonar-plugin: Execution default of goal org.apache.maven.plugins:maven-checkstyle-plugin:2.17:check failed: String index out of range: -1
```

I ca not reproduce it on local, removal of "~/.m2/repository/com/github" does not help.

@rnveach , do you remember how to reproduce it or find a reason or ... ?

**Comment by rnveach** [comment]
@romani This is https://issues.apache.org/jira/browse/MCHECKSTYLE-344 .
Issue occurs when checkstyle reports a violation on a module that has an `id` override defined.

**Comment by rnveach** [comment]
@romani Here is the raw XML output from my local:
````
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="8.5">
<file name="/media/checkstyle/work/sevntu.checkstyle/sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionPlugin.java">
<error line="3" severity="error" message="Line does not match expected header line of &apos;// Copyright (C) 2001-2018 the original author or authors.&apos;." source="header"/>
</file>
<file name="/media/checkstyle/work/sevntu.checkstyle/sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/package-info.java">
<error line="3" severity="error" message="Line does not match expected header line of &apos;// Copyright (C) 2001-2018 the original author or authors.&apos;." source="header"/>
</file>
<file name="/media/checkstyle/work/sevntu.checkstyle/sevntu-checkstyle-sonar-plugin/src/main/java/com/github/sevntu/checkstyle/sonar/CheckstyleExtensionRepository.java">
<error line="3" severity="error" message="Line does not match expected header line of &apos;// Copyright (C) 2001-2018 the original author or authors.&apos;." source="header"/>
</file>
<file name="/media/checkstyle/work/sevntu.checkstyle/sevntu-checkstyle-sonar-plugin/src/main/resources/com/github/sevntu/checkstyle/sonar/checkstyle-extensions.xml">
</file>
</checkstyle>
````

File: .travis.yml [hunk 1]
```diff
@@ -19,94 +19,49 @@ matrix:
...
-        - CMD0="cd sevntu-checks && mvn -e clean install -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dcobertura.skip=true && cd .. "
-        - CMD=$CMD0" && ""cd sevntu-checkstyle-idea-extension && mvn -e verify && mvn -e javadoc:javadoc"
+        - CMD="./.ci/travis.sh idea-extension"
 
     # sonar-plugin
     - jdk: oraclejdk8
       env:
         - DESC="sonar-plugin"
-        - CMD0="cd sevntu-checks && mvn -e clean install -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dcobertura.skip=true && cd .. "
-        - CMD=$CMD0" && ""cd sevntu-checkstyle-sonar-plugin && mvn -e verify && mvn -e javadoc:javadoc"
```
Comment (rnveach) [inline .travis.yml]: @romani is it possible this `" && ""` was causing a bug for the files not to be validated?

Comment (romani) [reply .travis.yml]: By shell syntax it looks right - just string concatenation

Comment (romani) [reply .travis.yml]: Looks like you about right , https://travis-ci.org/sevntu-checkstyle/sevntu.checkstyle/jobs/324756666#L441 ,  and https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/.travis.yml#L48 , command ci config were cut, partly executed.

**Comment by romani** [comment]
manually merged

**Pull request closed without merging**